lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2542,6 +2542,102 @@ class ConvertAtenOnesOp : public OpConversionPattern<AtenOnesOp> {
 };
 } // namespace
 
+namespace {
+
+static Type getTypeFromDTypeInteger(MLIRContext *context, int64_t dtypeInt) {
+  // TODO: include c10/core/ScalarType.h to make this cleaner.
+  switch (dtypeInt) {
+  case 6:
+    return Float32Type::get(context);
+  case 4:
+    return IntegerType::get(context, 64, IntegerType::Signed);
+  case 11:
+    return IntegerType::get(context, 1);
+  default:
+    return Type();
+  }
+}
+
+class ConvertAtenToDtypeOp : public OpConversionPattern<AtenToDtypeOp> {
+  using OpConversionPattern<AtenToDtypeOp>::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenToDtypeOp op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    AtenToDtypeOp::Adaptor adaptor(operands);
+
+    if (!adaptor.memory_format().getType().isa<Torch::NoneType>())
+      return rewriter.notifyMatchFailure(
+          op, "only default memory format is supported");
+
+    Location loc = op.getLoc();
+    Value dtype = op.dtype();
+    Value inVector = adaptor.self();
+    auto inType = inVector.getType().cast<RankedTensorType>();
+    int64_t inputRank = inType.getRank();
+    auto outType = getTypeConverter()
+                       ->convertType(op->getResult(0).getType())
+                       .cast<RankedTensorType>();
+
+    int64_t scalarType;
+    if (!matchPattern(dtype, m_TorchConstantInt(&scalarType))) {
+      return op.emitError("`dtype must be a constant integer");
+    }
+    Type outputType = getTypeFromDTypeInteger(op.getContext(), scalarType);
+
+    SmallVector<Value> outputDims;
+    for (auto i = 0; i < inputRank; i++)
+      outputDims.push_back(getDimOp(rewriter, loc, inVector, i));
+
+    Value outVector =
+        rewriter.create<linalg::InitTensorOp>(loc, outputDims, outputType);
+    SmallVector<AffineExpr> idExprs;
+    for (auto i = 0; i < inputRank; i++)
+      idExprs.push_back(getAffineDimExpr(i, rewriter.getContext()));
+    SmallVector<AffineMap> indexingMaps = {
+        AffineMap::get(inputRank, 0, idExprs, op.getContext()),
+        AffineMap::get(inputRank, 0, idExprs, op.getContext())};
+    SmallVector<StringRef> iteratorTypes(inputRank, "parallel");
+    auto result =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, outVector.getType(), inVector, outVector, indexingMaps,
+                iteratorTypes,
+                [](OpBuilder &b, Location loc, ValueRange args) {
+                  Value input = args[0];
+                  Type inType = input.getType();
+                  Type outType = args[1].getType();
+                  Value final;
+                  if (inType.isF32() || inType.isF64()) {
```

Comment:
We can simplify the types to only consider floating point type for now.

Reply:
Yes I have done that. Thanks!